components/carousel/index.jsx

```diff
@@ -0,0 +1,287 @@
+							onClick={this.onAutoPlayBtnClick}
+						/>
+					)}
+					<div className="slds-grid slds-grid_align-center">
+						{hasPreviousNextPanelNavigation && (
+							<PreviousNextCarouselNavigator
+								icon="/assets/icons/utility-sprite/svg/symbols.svg#left"
+								assistiveText={this.props.assistiveText.previousPanel}
+								isDisabled={isPreviousBtnDisabled}
+								onClick={this.onPreviousPanelHandler}
+							/>
+						)}
+						<div
+							ref={this.stageItem}
+							className="slds-carousel__stage slds-align_absolute-center"
+						>
+							<div
+								className="slds-carousel__panels slds-is-relative"
+								style={{
+									transform: `translateX(${this.state.translateX}px)`,
+								}}
+							>
+								{this.props.items.map((item) => (
+									<CarouselItem
+										onRenderItem={this.props.onRenderItem}
+										{...item}
+										itemWidth={itemWidth}
+										key={item.id}
+									/>
+								))}
+							</div>
+						</div>
+						{hasPreviousNextPanelNavigation && (
+							<PreviousNextCarouselNavigator
+								icon="/assets/icons/utility-sprite/svg/symbols.svg#right"
+								assistiveText={this.props.assistiveText.nextPanel}
+								isDisabled={isNextBtnDisabled}
+								onClick={this.onNextPanelHandler}
+							/>
+						)}
+					</div>
+					<CarouselIndicators
+						style={this.props.indicatorStyles}
+						noOfIndicators={this.nrOfPanels}
+						currentIndex={this.state.currentPanel}
+						onClick={this.onIndicatorClickHandler}
+					/>
+				</div>
+			</div>
+		);
+	}
+}
+
+Carousel.displayName = CAROUSEL;
+
+Carousel.propTypes = {
+	/**
+	 * Array of objects with shape, needed for building a carousel item.
+	 * A shape would have:
+	 *   - id: 0   [REQUIRED]
+	 *   - heading: "All my things! Part 0" // (Visible heading for default render)
+	 *   - description: "Card 0 description."
+	 *          // (Visible paragraph text for default render and navigation bullet text used in
+	 *              assistive text tag)   [REQUIRED]
+	 *   - imageAssistiveText: "First card accessible description."
+	 *          // (image alt text, OPTIONAL, required if `src` is present),
+	 *   - href: "https://www.salesforce.com" // (OPTIONAL, void(0) if not provided for default template
+	 *   - src: "/images/examples/carousel-01.png"
+	 */
+	items: PropTypes.array.isRequired,
+	/**
+	 * Number of items to be displayed at a time in the carousel
+	 */
+	itemsPerPanel: PropTypes.number,
+	/**
+	 * Boolean showing whether the autoplay feature is available or not
+	 */
+	hasAutoplay: PropTypes.bool,
+	/**
+	 * Interval for the autoplay iteration
+	 */
+	autoplayInterval: PropTypes.number,
+	/**
+	 * Boolean for displaying the navigation indicators (left/right arrows) of the carousel
+	 */
+	hasPreviousNextPanelNavigation: PropTypes.bool,
+	/**
+	 * Boolean for infinite loop navigation
+	 */
+	isInfinite: PropTypes.bool,
+	/**
+	 * CSS classes that are applied to the component
```

Comment:
Please mention the node they are applied to--that is `.slds-carousel`. I'm not seeing `this.props.className` anywhere though

Reply:
done